File: linear.java
class linear {
    public static int search(int arr[], int x)
    {
        int N = arr.length;
        for (int i = 0; i < N; i++) {
            if (arr[i] == x)
                return i;
        }
        return -1;
    }

    public static void main(String args[])
    {
        int arr[] = { 2, 3, 4, 10, 40 };
        int x = 3;
        System.out.print("[*] The elements of the array are: ");
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
        System.out.println("[*] The element to be searched is: " + x);
        int result = search(arr, x);
        if (result == -1)
            System.out.print(
                "[!] Element is not present in array\n");
        else
            System.out.print("[+] Element is present at index "
                             + result + ".\n");
    }
}

File: linear.c
#include <stdio.h>
int linearSearch(int arr[], int n, int val) {
  for (int i = 0; i < n; i++)
    {
        if (arr[i] == val)
        return i+1;
    }
  return -1;
}
int main() {
  int arr[] = {70, 40, 30, 11, 57, 41, 25, 14, 52};
  int val = 41;
  int n = sizeof(arr) / sizeof(arr[0]);
  int res = linearSearch(arr, n, val);
  printf("[*] The elements of the array are: ");
  for (int i = 0; i < n; i++)
  printf("%d ", arr[i]);
  printf("\n[*] The element to be searched is: %d", val);
  if (res == -1)
  printf("\n[!] Element is not present in the array");
  else
  printf("\n[+] Element is present at index %d.\n", res);
  return 0;
}
